ml-dhs-aws/src/main/java/com/marklogic/client/dhs/JobRunner.java

```diff
@@ -64,20 +95,42 @@ public void run(DatabaseClient client, InputStream csvRecords, InputStream jobCo
                         throwable.printStackTrace(System.out);
                      });
 
-         ObjectMapper objectMapper = new ObjectMapper();
-         ObjectNode jobControlObj = (ObjectNode) objectMapper.readTree(jobControl);
-         ObjectNode jobDef        = (ObjectNode) jobControlObj.get("job");
-         ObjectNode recordDef     = (ObjectNode) jobControlObj.get("record");
 
          ObjectLoader loader = new ObjectLoader(batcher, recordDef, getJobDirectory(), documentMetadata);
          CSVConverter converter = new CSVConverter();
 
-// TODO: write before job document in /jobStart and /jobs/ID collections or send before job payload to DHF endpoint
 
+
+         Iterator<ObjectNode> itr = converter.convertObject(csvRecords);
+         if(!itr.hasNext()) {
+        	 throw new MarkLogicIOException("No header found.");
+         }
+         ObjectNode csvNode = itr.next();
+         
+         Iterator<String> csvHeader = csvNode.fieldNames();
+         String headerValue = csvNode.get(csvHeader.next()).toString();
+         
+   // TODO: write before job document in /jobStart and /jobs/ID collections or send before job payload to DHF endpoint        
+         String beforeDocId = "/jobs/"+id+"/beforeJob.json";
+         
+         DocumentMetadataHandle beforeDocumentMetadata = new DocumentMetadataHandle();
+         beforeDocumentMetadata.withCollections("/jobs/"+beforeDocId, "/beforeJob");
          JobTicket ticket = moveMgr.startJob(batcher);
-         loader.loadRecords(converter.convertObject(csvRecords));
+         loader.loadRecord(csvNode);
+         
+         loader.loadRecords(itr);
          batcher.flushAndWait();
          moveMgr.stopJob(ticket);
+         
+         ObjectNode beforeDocRoot = objectMapper.createObjectNode();
+         beforeDocRoot.put("job id", id);
+         beforeDocRoot.put("reconciliation",  reconciliation);
```

Comment:
Instead, let's add a metadata key whose value is the job metadata node. 

Apologies for the muddy guidance on this point.